rdt-open-reader/src/main/java/io/ona/rdt/util/RDTJsonFormUtils.java

```diff
@@ -239,6 +239,7 @@ public void startJsonForm(JSONObject form, Activity context, int requestCode) {
         intent.putExtra(JSON_FORM_PARAM_JSON, form.toString());
         intent.putExtra(PERFORM_FORM_TRANSLATION, true);
         context.startActivityForResult(intent, requestCode);
+        RDTApplication.getInstance().clearCurrActivityReference(context);
```

Comment:
Why is this cleared here?

Reply:
now, form loading completed and we are starting the form activity so we don't need to persist source activity.